core/templates/dev/head/services/playthrough-issues-backend-api.service.ts

```diff
@@ -30,7 +30,7 @@ angular.module('oppia').factory('PlaythroughIssuesBackendApiService', [
       UrlInterpolationService, FETCH_ISSUES_URL, FETCH_PLAYTHROUGH_URL,
       RESOLVE_ISSUE_URL) {
     /** @type {PlaythroughIssue[]} */
-    var cachedIssues = null;
+    var cachedIssues = [];
```

Comment:
I'm not sure this change is correct. 

Suppose there are indeed 0 cached issues (since there may legitimately not be any issues reported for the exploration). Won't the changes made here result in continual refetching of the issues list, even though this is not necessary?

Note that `null` indicates "we haven't fetched anything yet" and `[]` indicates "we have fetched stuff and the list is empty, let's store that info here so we don't need to fetch it again later".

Reply:
I had changed `null` to `[]` because some methods were using `cachedIssues.length` or `cachedIssues.findIndex` when `cachedIssues` were `null`. I think what you said makes totally sense but I'll need to put some checking so these methods will be called only when `cachedIssue` is an array.